src/Authenticator/PasetoAuthenticator.php

```diff
@@ -0,0 +1,219 @@
+                throw new RuntimeException('PASETO `secretKey` config must be defined');
+            }
+            $this->setConfig('secretKey', \Cake\Utility\Security::getSalt());
+        }
+    }
+
+    /**
+     * Authenticates the identity based on a PASETO token contained in a request.
+     *
+     * @link https://paseto.io/
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request that contains login information.
+     * @return \Authentication\Authenticator\ResultInterface
+     * @throws \Exception
+     */
+    public function authenticate(ServerRequestInterface $request): ResultInterface
+    {
+        try {
+            $result = $this->getPayload($request);
+        } catch (Exception $e) {
+            return new Result(
+                null,
+                Result::FAILURE_CREDENTIALS_INVALID,
+                [
+                    'message' => $e->getMessage(),
+                    'exception' => $e,
+                ]
+            );
+        }
+
+        if (!$result instanceof JsonToken) {
+            return new Result(null, Result::FAILURE_CREDENTIALS_INVALID);
+        }
+
+        if (empty($result->getSubject())) {
+            return new Result(null, Result::FAILURE_CREDENTIALS_MISSING);
+        }
+
+        if ($this->getConfig('returnPayload')) {
+            $array = array_merge(
+                $result->getClaims(),
+                ['footer' => $result->getFooterArray()]
+            );
+
+            return new Result(new ArrayObject($array), Result::SUCCESS);
+        }
+
+        $user = $this->_identifier->identify([
+            'sub' => $result->getSubject(),
+        ]);
+
+        if (empty($user)) {
+            return new Result(null, Result::FAILURE_IDENTITY_NOT_FOUND, $this->_identifier->getErrors());
+        }
+
+        return new Result($user, Result::SUCCESS);
+    }
+
+    /**
+     * Get payload data.
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface|null $request Request to get authentication information from.
+     * @return object|null Payload object on success, null on failure
+     * @throws \Exception
+     */
+    public function getPayload(?ServerRequestInterface $request = null): ?object
+    {
+        if (!$request) {
+            return $this->payload;
+        }
+
+        $payload = null;
+        $token = $this->getToken($request);
+
+        if ($token !== null) {
+            $payload = $this->decodeToken($token);
+        }
+
+        $this->payload = $payload;
+
+        return $this->payload;
+    }
+
+    /**
+     * Decode PASETO token.
+     *
+     * @param string $token PASETO token to decode.
+     * @return null|\ParagonIE\Paseto\JsonToken The PASETO payload as a JsonToken object, null on failure.
+     * @throws \Exception
+     */
+    protected function decodeToken(string $token): ?JsonToken
+    {
+        if ($this->getConfig('purpose') === self::PUBLIC) {
+            $receivingKey = AsymmetricSecretKey::fromEncodedString(
```

Comment:
This seems like a reasonable default `purpose` to me.

Reply:
I was thinking LOCAL should be the default purpose. Local is similar to the default for JWT which uses an HMAC secret.